LayoutTests/imported/w3c/web-platform-tests/html/links/redirect-301-with-fragments.html

```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+    <script src='/resources/testharness.js'></script>
+    <script src='/resources/testharnessreport.js'></script>
+    <title>Ensure fragments are correctly transferred on redirects</title>
+</head>
+<body>
+    <script>
+        
```

Comment:
nit: extra line

Reply:
Also, we need 2 tests, one to cover the disk cache change and the other to cover the memory cache one.